clang/docs/checkedc/rfc/Include-Checked-Hdrs.md

```diff
@@ -0,0 +1,187 @@
+     supported. 5) In most cases integration into a build system will be easy:
+     the `CC` variable can be redefined appropriately.
+     - **Cons**: 1) It is a very heavy-weight solution. 2) In some cases
+     integrating with a build system will be more involved when both the drivers
+     are needed to compile the sources. This may happen in a scenario where most
+     source files want implicit inclusion but a few source files want explicit
+     inclusion because they want to directly include some system header files in
+     order to avoid Checked-C-specific declarations.
+
+ - Solution 3:
+     - Let `foo.h` be a system header file with its checked counterpart called
+       `foo_checked.h`. We add a new file called `foo.h` in the same directory
+       that contains `foo_checked.h`. In this new file, we add the following:
+
+             #ifdef IMPLICIT_INCLUDE_CHECKED_HDRS
+             #include <foo_checked.h>
+             #else
+             #include_next <foo.h>
+             #endif
+     - In `foo_checked.h`, we modify `#include <foo.h>` to
+       `#include_next <foo.h>`.
+
+     - The way this solution will work:
+        - If a program includes `foo.h`, clang will pick up either
+          `foo_checked.h` or the system `foo.h` depending on whether or not
+          `-DIMPLICIT_INCLUDE_CHECKED_HDRS` is specified on the compilation
+          commandline.  Therefore specifying this flag will cause the implicit
+          inclusion of the checked counterpart of `foo.h`, which will make it
+          convenient for automation. Not specifying the commandline flag will
+          not perform implicit inclusion, satisfying our "opt-in" philosophy.
+          Infinite recursion is avoided because of `#include_next`.
+        - If a program includes `foo_checked.h` current behavior will prevail.
+          The above flag has no effect on the explicit inclusion of checked
+          header files.
+
+     - **Pros**: 1) It satisfies the "opt-in" philosophy. 2) It is convenient
+       for automation. 3) It provides fine-grained control to directly include a
+       system header file. 4) The existing approach of explicit inclusion is
+       supported. 5) In most cases integration with the build system is easy:
+       the flag `-DIMPLICIT_INCLUDE_CHECKED_HDRS` is passed to the compilation
+       of all source files through the `CFLAGS` variable.
+
+     - **Cons**: 1) In some cases integration with the build system will be more
+       involved if the above flag needs to be passed to the compilation of most
+       source files and avoided for a few source files that may want to directly
+       include system header files in order to avoid Checked-C-specific
+       declarations. 2) It is difficult to accomodate clang-specific
+       declarations.
+
+## Proposed Solution:
+
+The proposed solution is as follows:
+  - It is a hybrid solution based on ideas in Solutions 3 and 1 that
```

Comment:
Shouldn't t his be classified as another possible solution?  It is a little confusing for the proposed solution to be described as a hybrid of other solutions.